projects/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-listener.php

```diff
@@ -79,6 +79,11 @@ function test_detects_if_exceeded_queue_size_limit_and_oldest_item_gt_15_mins()
 		do_action( 'my_action' );
 		$this->assertEquals( 3, $this->listener->get_sync_queue()->size() );
 
+		// reset queue settings.
+		Settings::update_settings( array( 'max_queue_size' => Defaults::$default_max_queue_size ) );
+		Settings::update_settings( array( 'max_queue_lag' => Defaults::$default_max_queue_lag ) );
+		$this->listener->set_defaults(); // should reset queue size limit.
```

Comment:
This fix works as long as the test is passing. If one of the assertions in the test fails, though, the cleanup won't happen. That would mean a failure in this test would make all those later tests start failing again too, making things seem more dire than they actually are.

One useful pattern for that is a try-finally:
```php
	// set max queue size to 2 items
	Settings::update_settings( array( 'max_queue_size' => 2 ) );

	// set max queue age to 3 seconds
	Settings::update_settings( array( 'max_queue_lag' => 3 ) );

	try {

		// do stuff, calling $this->assertEquals() and the like

	} finally {
		// Reset settings before returning.
		Settings::update_settings( array( 'max_queue_size' => Defaults::$default_max_queue_size ) );
		Settings::update_settings( array( 'max_queue_lag' => Defaults::$default_max_queue_lag ) );
		$this->listener->set_defaults(); // should reset queue size limit.
	}
```